featurebyte/api/feature_or_target_mixin.py

```diff
@@ -0,0 +1,73 @@
+"""
+Mixin class containing common methods for feature or target classes
+"""
+from typing import cast
+
+from featurebyte.api.api_object import ApiObject
+from featurebyte.common.formatting_util import CodeStr
+from featurebyte.common.typing import Func
+from featurebyte.core.generic import QueryObject
+from featurebyte.exception import RecordRetrievalException
+from featurebyte.models.feature import BaseFeatureModel
+
+DEFINITION_DOCSTRING = """
+Displays the {object_type} definition file of the {object_type}.
+
+The file is the single source of truth for a {object_type} version. The file is generated automatically after a
+{object_type} is declared in the SDK and is stored in the FeatureByte Service.
+
+This file uses the same SDK syntax as the {object_type} declaration and provides an explicit outline of the intended
+operations of the {object_type} declaration, including those that are inherited but not explicitly declared by the
+user. These operations may include feature job settings and cleaning operations inherited from tables metadata.
+
+The {object_type} definition file serves as the basis for generating the final logical execution graph, which is
+then transpiled into platform-specific SQL (e.g. SnowSQL, SparkSQL) for {object_type} materialization.
+
+Returns
+-------
+str
+
+Examples
+--------
+{example}
+"""
+
+
+def substitute_docstring(new_docstring: str) -> Func:
+    """
+    Decorator to substitute the docstring of a function
+
+    Parameters
+    ----------
+    new_docstring: str
+        New docstring to use
+
+    Returns
+    -------
+    Func
+    """
+
+    def decorator(func: Func) -> Func:
```

Comment:
Tested it locally, seems like it will void the type checking. Since this is used mostly at the SDK api object public method, the impact should be less severe than those internal methods. If this is a serious concern and there is no other way to get around this, we may need to look for other approach to share the doc string.

Reply:
I created a test method at `Feature` object:
```
    def test(self) -> int:
        return self.definition
```

I get the following error when running mypy:
![Screenshot 2023-07-11 at 3 55 13 PM](https://github.com/featurebyte/featurebyte/assets/37915749/1ac7580b-8697-43b8-a289-7f59c5912764)

Seems like it still can detect the error.

Reply:
I see, seems like it's a non-issue 👍